File: leetcode/트리/938/sol.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    answer : int = 0
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        
        def dfs(root) : 
            if root is None : 
                return
            
            if root.val <= high and root.val >= low : 
                self.answer += root.val
                
            dfs(root.left)
            dfs(root.right)
            
        dfs(root)
        
        return self.answer

File: leetcode/트리/110/sol.cpp
class Solution {
public:
    int check(TreeNode* node) {
        if(node == nullptr) {
            return 0;
        }
        
        int left = check(node->left);
        int right = check(node->right);
        
        if(left == -1 || right == -1 || abs(left - right) > 1) {
            return -1;
        }
        
        return max(left, right) + 1;
    }
    
    bool isBalanced(TreeNode* root) {
        return check(root) != -1;
    }
};